pkg/apis/core/v1alpha1/extensionrepo_types.go

```diff
@@ -68,11 +68,24 @@ type ExtensionRepoSpec struct {
 	// the repo to the latest version.
 	// +optional
 	Ref string `json:"ref,omitempty" protobuf:"bytes,2,opt,name=ref"`
+
+	// A prefix used to support autoregistration of extensions within this repository.
+	// For example, an extension repo with prefix="internal" will cause any load()
+	// calls that start with `ext://internal` to be loaded from this repository.
+	// +optional
+	Prefix string `json:"prefix,omitempty" protobuf:"bytes,3,opt,name=prefix"`
```

Comment:
how do you feel about naming these as:

LoadHost -- used for `load("ext://{LoadHost}/path/to/extension")`
GitSubPath -- subpath in a git repo to find the extension, only valid when URL points to a git repo

i 100% agree that naming and ergonomics is the hardest part!!

Reply:
re: GitSubPath

`GitSubPath` works in the general case where the extension repo is an actual git repository, but feels kind of misnamed when you're using a local extension repo. Although I guess in that case you wouldn't need the extra path specifier either (since you can just specify it in the repo path itself).

Maybe there's an opportunity to extend the "url" part to carry a subpath as well? Naively you can do `<host>/<owner>/<repo>/<path>` but that fails for hosts like gitlab or bitbucket where there's often an extra path component already for a "project". To handle that the resolver *could* have special casing for the most common code hosts with a fallback that attempts cloning starting from the full url, chopping off a path segment on each fetch failure until it gets to the base case of `<host>/<repo>` but that seems maybe too convoluted.

I also feel like `extensions_path` matches more closely with `repo_path` in the `extension` API and might be more clear of a name. I think `subpath` gets most of the way there but mostly for whoever is writing the `extension_repo` line in a `Tiltfile`, it may not be clear enough for the other people on the team that are just *reading* the line.

re: LoadHost

I'm not sure what `Host` is meant to indicate. Now that I'm thinking about it more, this could theoretically be simplified a little bit if, instead of a string prefix, this is a boolean that defaults to false, and if true, registers the name of the repository as the "load prefix".

So, something like `extension_repo(name='datadog', url='...', register_extensions=True)` would trigger auto registration for any otherwise unregistered extension that starts with `ext://datadog/`.

Reply:
re: what is Host meant to indicate - https://datatracker.ietf.org/doc/html/rfc1738#section-3.1, more as a convention, because it's that part of the URI

re: ya, i think we'd want it to be an error if you use GitSubPath and a file URL.

Reply:
and ya, i'd rather avoid sticking more special cases in the URL and lean towards explicit fields

Reply:
> re: what is Host meant to indicate - https://datatracker.ietf.org/doc/html/rfc1738#section-3.1, more as a convention, because it's that part of the URI

ohhh, i wasn't even thinking about it as "hostname".

Ultimately I'm happy to change it to whatever you prefer. Do you want `GitSubPath / git_subpath` and `LoadHost / load_host`?

Reply:
yes, thank you!

Reply:
done in the latest commit(s)